neqo-http3/src/stream_type_reader.rs

```diff
@@ -52,7 +53,16 @@ impl NewStreamType {
             | (WEBTRANSPORT_UNI_STREAM, StreamType::UniDi, _)
             | (WEBTRANSPORT_STREAM, StreamType::BiDi, _) => Ok(None),
             (H3_FRAME_TYPE_HEADERS, StreamType::BiDi, Role::Server) => Ok(Some(Self::Http)),
-            (_, StreamType::BiDi, Role::Server) => Err(Error::HttpFrame),
+            (stream_type, StreamType::BiDi, Role::Server) => {
+                let Some(st) = stream_type.checked_sub(0x21) else {
+                    return Err(Error::HttpFrame);
+                };
+                if st % 0x1f == 0 {
+                    Ok(Some(Self::Grease(stream_type)))
+                } else {
+                    Err(Error::HttpFrame)
+                }
```

Comment:
As we discussed, this is not correct.

For all BiDi streams, as long as the type is not WEBTRANSPORT_STREAM, then we should be treating this as OK.  ... unless the frame type is one we understand other than HEADERS.

This will unnecessarily catch a new extension frame type that is added ahead of HEADERS.